src/Compilers/VisualBasic/Portable/Lowering/AsyncRewriter/AsyncRewriter.AsyncMethodToClassRewriter.Spilling.vb

```diff
@@ -310,7 +316,13 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                             Return fieldAccess
                         End If
 
-                        Dim newReceiver As BoundExpression = SpillValue(fieldAccess.ReceiverOpt, isReceiver:=True, builder:=builder)
+                        Dim checkSideEffectsHere = shouldCheckSideEffects And Not sideEffectsAlreadyChecked
+
+                        Dim newReceiver As BoundExpression = SpillValue(fieldAccess.ReceiverOpt,
```

Comment:
>Dim newReceiver As BoundExpression = SpillValue(fieldAccess.ReceiverOpt, [](start = 24, length = 72)

Suppose we are assigning to ```a.b.c``` where ```a``` is a local, ```b``` and ```c``` are fields in structures.
First, we get here for ```c``` with shouldCheckSideEffects:=False, sideEffectsAlreadyChecked:=False, ```checkSideEffectsHere``` is False, side-effects are not evaluated below
Then, we get here for ```b``` with shouldCheckSideEffects:=True, sideEffectsAlreadyChecked:=False, ```checkSideEffectsHere``` is True, side-effects are going to be evaluated below
Do we really need to evaluate them? Why? What am I missing?